src/ucs/sys/sock.c

```diff
@@ -14,15 +14,28 @@
 #include <unistd.h>
 #include <errno.h>
 #include <string.h>
+#include <sys/uio.h>
+/* need this to get IOV_MAX on some platforms. */
+#ifndef __need_IOV_MAX
+#define __need_IOV_MAX
+#endif
 #include <limits.h>
 
 
 #define UCS_SOCKET_MAX_CONN_PATH "/proc/sys/net/core/somaxconn"
 
+/* Maximum length of the 'struct iovec' array in a single call to
+ * readv/writev/sendmsg/recvmsg/etc.
+ * The value used as a fallback when system value is not available.
+ * The latest kernels define it as 1024 */
+#define UCS_SOCKET_MAX_IOV       1024
```

Comment:
WDYT about moving this inside ucs_socket_max_iov()?
without defining UCS_SOCKET_MAX_IOV macor?

Reply:
done